shared/macros-ansible.jinja

```diff
@@ -0,0 +1,54 @@
+{{#
+  A wrapper over the Ansible lineinfile module. This handles the most common
+  options for us. regex is optional and when blank, it won't be included in
+  the Ansible script; this allows arbitrary additions to files. new_line will
+  only be passed when state is present. with_items will be specified only if
+  non-empty, allowing for iterating through a variable of content (with the
+  appropriate macro-based path). register will be specified only if non-empty,
+  allowing for saving the output of this lineinfile module.
+
+  Note that all string-like parameters are single quoted in the YAML.
+#}}
+{{% macro ansible_simple_line_in_file(msg='', path='', regex='', new_line='', create='no', state='present', with_items='', register='') -%}}
+- name: "{{{ msg }}}"
+  lineinfile:
+    path: '{{{ path }}}'
+    create: {{{ create }}}
+    {{%- if regex %}}
+    regexp: '{{{ regex }}}'
+    {{%- endif %}}
+    {{%- if state == 'present' %}}
+    line: '{{{ new_line }}}'
+    state: present
+    {{% else %}}
+    state: '{{{ state }}}'
+    {{%- endif %}}
+  {{%- if with_items %}}
+  with_items: '{{{ with_items }}}'
+  {{%- endif %}}
+  {{%- if register %}}
+  register: '{{{ register }}}'
+  {{%- endif %}}
+{{%- endmacro %}}
+
+{{#
+  A wrapper for adding one, unique line to a file. A regex must be specified
+  to tell if the line is unique. This is helpful in configuration files where
+  a single configuration parameter might have multiple values, but only one
+  value is approved. All lines matching the regex are first removed and then
+  the new line is appended to the file.
+#}}
+{{% macro ansible_one_line_in_file(msg, path, line_regex, new_line, create='no') -%}}
+{{{ ansible_simple_line_in_file("[DEDUPLICATE] " + msg, path, regex=line_regex, new_line=None, create='no', state='absent') }}}
+{{{ ansible_simple_line_in_file(msg, path, regex=None, new_line=new_line, create=create, state='present') }}}
```

Comment:
Another idea: This creates 2 independent Ansible tasks. Our build system adds tags to each task, so the output becomes little bloated (`build/rhel8/fixes/ansible/disable_host_auth.yml`) Do you think it would be feasible to use blocks (https://docs.ansible.com/ansible/latest/user_guide/playbooks_blocks.html) to merge the 2 tasks together?  I don't know if it's possible due to the indentation, but it might be worth investigating.

Reply:
Ah that'd be an interesting idea, I'll go and try it. :) Thanks @jan-cerny!

Reply:
@jan-cerny Jinja appears to provide a "indent" helper function so this is doable. :)